activation/atxdb.go

```diff
@@ -484,7 +465,7 @@ func (db *ActivationDb) GetAtx(id types.AtxId) (*types.ActivationTx, error) {
 	if err != nil {
 		return nil, err
 	}
-	atx, err := types.BytesAsAtx(b)
+	atx, err := types.BytesAsAtx(b, &id)
 	if err != nil {
 		return nil, err
 	}
```

Comment:
Not sure if it is a problem, but adding the atx to the cache without the lock might end up with to readers adding the same ATX twice

Reply:
Not an issue. I checked the code and the `Add()` method just promotes an entry if it already exists, so adding a value with the same key twice has no effect, in effect.